### ISSUE TITLE
Add "favorite" functionality

### ISSUE THREAD

**Issue by tedbarnett** [tedbarnett/VideoWatcher#2]
* Add a hollow "heart" icon visible in the top right of each video panel
* When tapped, heart is "filled" (red) and video is added to list of favorites

**Comment by tedbarnett** [comment]
As relates to Issue #4:
If a user taps the heart on any video under 30 seconds in length, the entire video is made into a "clip" (and the heart becomes filled red.  If longer, heart is filled pink, which means we have to edit the clips in full-screen mode later.

**Comment by bhautikpatoliya004** [comment]
Currently, I’ve added whole video as favorite on tap of Heart icon. And When tapped, heart is "filled" with Pink color. The favorite video will display on Menu > Favorite screen.
Can you please let me know how the Clip will work and how we will generate clip?

**Comment by tedbarnett** [comment]
For now, this is a good start.  Some videos are short enough to be favorites in their entirety, but I will later add a feature for longer videos (over 1 minute long) that asks if the user wants to select a shorter "clip" within the video.

**Comment by bhautikpatoliya004** [comment]
@tedbarnett Do you mean I need to show popup when tap on hollow "heart"? It will display message "Do you wants to select a shorter "clip" within the video" if video longer than 30 seconds?

**Comment by tedbarnett** [comment]
Yes.  It's a little clumsy but it's a start.  So: if tap heart on a video longer than 30 seconds, then show the dialog "this video is x:xx minutes long.  Do you want to favorite the entire video, or select a shorter clip within the video?".  Buttons: "Entire Video", "Shorter Clip", "Cancel".  If "Entire Video" is tapped, add heart to the entire video and close dialog.  If "Shorter Clip" selected: open the Full Screen view of the video and show the video clip selection option (NOT DEFINED YET).  If "Cancel", then close dialog and do not add heart to this video.
